packages/seedless-onboarding-controller/src/SeedlessOnboardingController.ts

```diff
@@ -1856,6 +1875,81 @@ export class SeedlessOnboardingController<EncryptionKey> extends BaseController<
     });
   }
 
+  /**
+   * Revoke all pending refresh tokens.
+   *
+   * This method is to be called after user is authenticated.
+   */
+  async revokePendingRefreshTokens() {
+    this.#assertIsAuthenticatedUser(this.state);
+    const { pendingToBeRevokedTokens } = this.state;
+    if (!pendingToBeRevokedTokens || pendingToBeRevokedTokens.length === 0) {
+      return;
+    }
+
+    // revoke all pending refresh tokens in parallel
+    const promises = pendingToBeRevokedTokens.map(({ revokeToken }) => {
+      const revokePromise = async () => {
+        try {
+          await this.#revokePendingRefreshToken(revokeToken);
+        } catch (error) {
+          log('Error revoking refresh token', error);
+        }
+      };
+      return revokePromise();
+    });
+    await Promise.all(promises); // no need to do Promise.allSettled because the promise already handle try catch
+  }
+
+  /**
+   * Revoke a single pending refresh token.
+   *
+   * @param revokeToken - The revoke token to revoke.
+   */
+  async #revokePendingRefreshToken(revokeToken: string) {
+    await this.#revokeRefreshToken({
+      connection: this.state.authConnection as AuthConnection,
+      revokeToken,
+    });
+    this.#removePendingRefreshToken({ revokeToken });
+  }
+
+  /**
+   * Add a pending refresh, revoke token to the state to be revoked later.
+   *
+   * @param params - The parameters for adding a pending refresh, revoke token.
+   * @param params.refreshToken - The refresh token to add.
+   * @param params.revokeToken - The revoke token to add.
+   */
+  #addPendingRefreshToken({
+    refreshToken,
+    revokeToken,
+  }: {
+    refreshToken: string;
+    revokeToken: string;
+  }) {
+    this.update((state) => {
+      state.pendingToBeRevokedTokens = [
+        ...(state.pendingToBeRevokedTokens || []),
+        { refreshToken, revokeToken },
+      ];
+    });
+  }
+
+  /**
+   * Remove a pending refresh, revoke token from the state.
+   *
+   * @param params - The parameters for removing a pending refresh, revoke token.
+   * @param params.revokeToken - The revoke token to remove.
+   */
+  #removePendingRefreshToken({ revokeToken }: { revokeToken: string }) {
+    this.update((state) => {
+      state.pendingToBeRevokedTokens = state.pendingToBeRevokedTokens?.filter(
```

Comment:
we should acquire lock on state before adding or removing revoked tokens since we are calling the revocation of all tokens parrallely. It can cause list having stale data due to race conditions.